examples/texture-arrays/main.rs

```diff
@@ -97,29 +97,26 @@ impl framework::Example for Example {
         queue: &wgpu::Queue,
     ) -> (Self, Option<wgpu::CommandBuffer>) {
         let mut uniform_workaround = false;
-        let vs_bytes: &[u8] = include_bytes!("shader.vert.spv");
-        let fs_bytes: &[u8] = match device.capabilities() {
+        let vs_module = device.create_shader_module(wgpu::include_spirv!("shader.vert.spv"));
+        let fs_bytes: Vec<u8> = match device.capabilities() {
             c if c.contains(wgpu::Capabilities::UNSIZED_BINDING_ARRAY) => {
-                include_bytes!("unsized-non-uniform.frag.spv")
+                include_bytes!("unsized-non-uniform.frag.spv").to_vec()
             }
             c if c.contains(wgpu::Capabilities::SAMPLED_TEXTURE_ARRAY_NON_UNIFORM_INDEXING) => {
-                include_bytes!("non-uniform.frag.spv")
+                include_bytes!("non-uniform.frag.spv").to_vec()
             }
             c if c.contains(wgpu::Capabilities::SAMPLED_TEXTURE_ARRAY_DYNAMIC_INDEXING) => {
                 uniform_workaround = true;
-                include_bytes!("uniform.frag.spv")
+                include_bytes!("uniform.frag.spv").to_vec()
             }
             c if c.contains(wgpu::Capabilities::SAMPLED_TEXTURE_BINDING_ARRAY) => {
-                include_bytes!("constant.frag.spv")
+                include_bytes!("constant.frag.spv").to_vec()
             }
             _ => {
                 panic!("Graphics adapter does not support any of the capabilities needed for this example");
             }
         };
-        let vs_module =
-            device.create_shader_module(&wgpu::read_spirv(std::io::Cursor::new(vs_bytes)).unwrap());
-        let fs_module =
-            device.create_shader_module(&wgpu::read_spirv(std::io::Cursor::new(fs_bytes)).unwrap());
+        let fs_module = device.create_shader_module(wgpu::util::make_spirv(&fs_bytes));
```

Comment:
There is no guarantee that fs_bytes is aligned!

Reply:
It's going to be aligned to the alignment of the global memory allocator, which on all platforms we care about is > 4. Wouldn't hurt to add an assert here though.

Reply:
There already is an assert.

So heap memory is automatically aligned to 4.
But stack memory has no such automatic alignment?
I haven't heard of this before.

Reply:
Heap memory on linux is aligned to 16, windows 8, don't remember mac but it's probably one of them. Depends on the properties of the allocator, but it's often helpful inside the allocator to have a minimum size that you can allocate.